js/sia.js

```diff
@@ -158,7 +160,13 @@ function SiadWrapper () {
 
     // Spawn siad
     const Process = require('child_process').spawn
-    var daemonProcess = new Process(siad.fileName, processOptions)
+    var daemonProcess = new Process(siad.fileName, {
+      '--agent': siad.agent,
+      '--api-addr': siad.address,
+      '--rpc-addr': siad.rpcAddress,
+      '--host-addr': siad.hostAddress,
+      '--sia-directory': siad.path
+    }, processOptions)
```

Comment:
I've tested it just by running it with the UI, unit tests are a while away for the wrapper.